testsuites/syncgateway/functional/tests/test_log_redaction.py

```diff
@@ -0,0 +1,454 @@
+    ansible_runner = AnsibleRunner(cluster_config)
+
+    log_info("Pulling sync_gateway / sg_accel logs")
+    # fetch logs from sync_gateway instances
+    status = ansible_runner.run_ansible_playbook("fetch-sync-gateway-all-logs.yml")
+    if status != 0:
+        raise CollectionError("Could not pull logs")
+    temp_log_path = ""
+    # zip logs and timestamp
+    if os.path.isdir("/tmp/sg_logs"):
+        date_time = time.strftime("%Y-%m-%d-%H-%M-%S")
+        temp_log_path = "/tmp/{}-{}-sglogs".format("log-redaction", date_time)
+        shutil.copytree("/tmp/sg_logs", temp_log_path)
+        temp_log_path1 = "{}/sg1".format(temp_log_path)
+        temp_log_path_list = [temp_log_path1 + "/sg_info.log", temp_log_path1 + "/sync_gateway_error.log"]
+        if mode.lower() == "di":
+            temp_log_path_list.append(temp_log_path + "/ac1/sg_accel_error.log")
+            temp_log_path_list.append(temp_log_path + "/ac1/sg_debug.log")
+            temp_log_path_list.append(temp_log_path + "/ac1/sg_info.log")
+        for item in temp_log_path_list:
+            try:
+                scan_for_pattern(item, ["<ud>", "</ud>"])
+            except LogScanningError as le:
+                if log_redaction_level == "none":
+                    continue
+                else:
+                    assert False, le.message
+
+    # verify starting and ending ud tags are equal
+    num_ud_tags = subprocess.check_output("find {} -name '*.log' | xargs grep '<ud>' | wc -l".format(temp_log_path), shell=True)
+    num_end_ud_tags = subprocess.check_output("find {} -name '*.log' | xargs grep '</ud>' | wc -l".format(temp_log_path), shell=True)
+    assert num_ud_tags == num_end_ud_tags, "There is a mismatch of ud tags"
+    shutil.rmtree(temp_log_path)
+
+
+def sgcollect_redact(cluster_config, log_redaction_level, redaction_salt):
+    ansible_runner = AnsibleRunner(cluster_config)
+
+    log_info("started sg collect info")
+    # fetch logs from sync_gateway instances
+    date_time = time.strftime("%Y-%m-%d-%H-%M-%S")
+    zip_file_name = "sgcollect-{}".format(date_time)
+    salt_value = ""
+    if redaction_salt:
+        salt_value = "--log-redaction-salt='sg_collect test'"
+    status = ansible_runner.run_ansible_playbook(
+        "sgcollect-info.yml",
+        extra_vars={
+            "redact_level": log_redaction_level,
+            "zip_file_name": zip_file_name,
+            "salt_value": salt_value
+        }
+    )
+    if status != 0:
+        raise CollectionError("Could not pull logs")
+    return zip_file_name
+
+
+def pull_redacted_zip_file(cluster_config, sg_platform, output_dir=None):
+    ansible_runner = AnsibleRunner(cluster_config)
+    if output_dir is None:
+        if sg_platform == "centos":
+            sg_logs_dir = "/home/sync_gateway"
+            sa_logs_dir = "/home/sg_accel"
+        if sg_platform == "windows":
+            sg_logs_dir = "C:{}".format("\PROGRA~2\Couchbase\\var\\")
```

Comment:
can be hardcoded, no need for formatting

Reply:
In pythong we have problem that "\v" adds some characters while running the test, we need to add double slash . I had some problem in adding it path directly, so have to use format

Reply:
Should not be a problem with two backslashes
```
>>> sg_logs_dir = "C:\\PROGRA~2\Couchbase\\var\\"
>>> sg_logs_dir
'C:\\PROGRA~2\\Couchbase\\var\\'
```

Reply:
Yes ofcourse, first thing I used is without format, :\ combination makes mess when it uses in ansible, that is why I had to use format.